File: ios/MessageParser.swift
import Crisp
import Foundation

/**
 * Parser for converting native Crisp Message to JS-compatible dictionary.
 */
enum MessageParser {
  static func toDictionary(_ message: Message) -> [String: Any] {
    let textContent: String
    switch message.content {
    case .text(let text):
      textContent = text
    case .textWithAttachment(let text, _, _):
      textContent = text
    case .textWithVideoAttachment(let text, _, _):
      textContent = text
    default:
      textContent = ""
    }

    var dict: [String: Any] = [
      "content": textContent,
      "timestamp": message.timestamp.timeIntervalSince1970 * 1000,
      "fromOperator": message.from == .operator,
      "fingerprint": String(message.fingerprint)
    ]

    // Add user info if available
    if let user = message.user {
      var userDict: [String: Any] = [:]
      if let nickname = user.nickname {
        userDict["nickname"] = nickname
      }
      if let userId = user.userId {
        userDict["userId"] = userId
      }
      if let avatar = user.avatar {
        userDict["avatar"] = avatar.absoluteString
      }
      if !userDict.isEmpty {
        dict["user"] = userDict
      }
    }

    return dict
  }
}

File: android/src/main/java/expo/modules/crispsdk/MessageParser.kt
package expo.modules.crispsdk

import im.crisp.client.external.data.message.Message
import im.crisp.client.external.data.message.content.TextContent

/**
 * Parser for converting native Crisp Message to JS-compatible format.
 */
object MessageParser {
    fun toMap(message: Message): Map<String, Any?> {
        val content = message.content
        val textContent = if (content is TextContent) content.text else content?.toString() ?: ""

        return mapOf(
            "content" to textContent,
            "timestamp" to message.timestamp,
            "fromOperator" to (message.from == Message.From.OPERATOR)
        )
    }
}
